etcdctl/command/exec_watch_command.go

```diff
@@ -51,21 +51,35 @@ func execWatchCommandFunc(c *cli.Context, ki client.KeysAPI) {
 		handleError(ExitBadArgs, errors.New("key and command to exec required"))
 	}
 
-	key := args[argslen-1]
-	cmdArgs := args[:argslen-1]
+	var (
+		key     string
+		cmdArgs []string
+	)
+
+	foundSep := false
+	for i := range args {
+		if args[i] == "--" && i != 0 {
```

Comment:
ok. so either `$ etcdctl exec-watch /foo/bar -- sh -c "env | grep ETCD"` or `$ etcdctl exec-watch sh -c "env | grep ETCD" /foo/bar` is valid.